experimenter/query.py

```diff
@@ -0,0 +1,147 @@
+CONNECTION = Elasticsearch(hosts=[HOST], timeout=300)
+
+         
+def get_search_query(arguments: dict = None, connection = CONNECTION, index='pipelines'):
+    index_search = Search(using=CONNECTION, index=index)
+    if arguments['id'] is not None:
+        index_search = index_search.query('match', id=arguments['id'])
+    if arguments['submitter'] is not None:
+        index_search = index_search.query('match', _submitter=arguments['submitter'])
+    return index_search
+
+
+def query_on_seeds(pipeline_id: str=None, limit: int=None, submitter: str='byu'):
+    arguments = {'id': pipeline_id, 'submitter': submitter}
+    pipeline_search = get_search_query(arguments=arguments, index='pipelines')
+    for pipeline in pipeline_search.scan():
+        results = scan_pipeline_runs(pipeline.id, submitter)
+        for dataset_prob_tuple, results_dict in results.items():
+            unique_items = get_unique_results(results_dict)
+            #unpack values from tuple
+            problem_id, dataset_id, data_prep, scoring = dataset_prob_tuple
+            scoring_id, scoring_random_seed = scoring
+            data_prep_id, data_prep_seed = data_prep
+            #get preparation and scoring pipelines
+            data_prep_pipeline = get_pipeline(data_prep_id, types='Data')
+            scoring_pipeline = get_pipeline(scoring_id, types='Scoring')
+            for params in unique_items:
+                data_params = params['data_params']
+                scoring_params = params['scoring_params']
+                random_seeds = params['random_seeds']    
+                if limit and len(random_seeds) > limit:
+                    continue
+                yield {'pipeline': pipeline.to_dict(), 'problem_path': get_problem_path(problem_id), 
+                       'dataset_doc_path':get_dataset_doc_path(dataset_id), 'tested_seeds': random_seeds,
+                       'data_prep_pipeline': data_prep_pipeline, 'data_prep_seed': data_prep_seed,
+                       'scoring_pipeline': scoring_pipeline, 'scoring_seed': scoring_random_seed,
+                       'scoring_params': scoring_params, 'data_params': data_params}
+
+
+def get_pipeline(pipeline_id: str=None, types: str='Data'):
+      if (pipeline_id is None):
+          return None
+      pipeline = get_pipelines_from_d3m(pipeline_id, types=types)
+      #get from database if not in d3m module
+      if (pipeline is None):
+          arguments = {'submitter': None, 'id': data_prep_id}
+          search = get_search_query(arguments=arguments)
+          pipeline = next(search.scan())
+          pipeline = pipeline.to_dict()
+      return pipeline
+
+
+def check_for_data_prep(pipeline_run=None):
+    """Only handles cases with an explicit data preparation pipeline
+    in the pipeline run
+    """
+    data_prep = None
+    data_prep_id = None
+    data_prep_seed = None
+    data_params = None
+    try:
+        data_prep = pipeline_run.run.data_preparation
+    except:
```

Comment:
Specify the exception you are trying to catch.

Reply:
How do you want me to do that?

Reply:
https://docs.python.org/3/tutorial/errors.html#handling-exceptions